include/picongpu/fields/laserProfiles/PulseFromSpectrum.hpp

```diff
@@ -0,0 +1,335 @@
+                        float_X const runTime = float_X(currentStep - N) * DELTA_T;
+
+                        /* Calculation of the E(t) using trigonometric Interpolation.
+                         * Coefficients can be easily determined cause the spectrum is given.
+                         */
+                        float_64 E_a = (1._X / DELTA_T) * transversal_spectrum(0._X, r2, y_to_focus)
+                            * math::cos(phase_v(0._X, r2, y_to_focus)); // from first symm. coeff. (E_a(k=0) = a/2)
+                        float_64 E_b = 0._X; // from first antisymm. coeff. (E_b(k=0) = 0)
+
+                        // rest of trig. Interpolation
+                        for(int k = 1; k < N + 1; ++k)
+                        {
+                            float_X v_k = float_X(k) / (float_X(2 * N + 1) * DELTA_T); // discrete values of frequency
+                            float_X a = (2._X / DELTA_T) * transversal_spectrum(v_k, r2, y_to_focus)
+                                * math::cos(phase_v(v_k, r2, y_to_focus)); // symm. coeff. trig. Interpolation
+                            float_X b = (2._X / DELTA_T) * transversal_spectrum(v_k, r2, y_to_focus)
+                                * math::sin(phase_v(v_k, r2, y_to_focus)); // antisymm. coeff. trig. Interpolation
+
+                            E_a += a * math::cos(2._X * float_X(PI) * v_k * runTime);
+                            E_b += b * math::sin(2._X * float_X(PI) * v_k * runTime);
+                        }
+
+                        // E(t)-Field derived from Spectrum
+                        float_64 E_t = (E_a + E_b) / float_X(2 * N + 1);
+                        E_t *= Unitless::AMPLITUDE;
+
+                        return E_t;
+                    }
+
+                    /** Device-Side Constructor
+                     *
+                     * @param superCellToLocalOriginCellOffset local offset in cells to current supercell
+                     * @param offsetToTotalDomain offset to origin of global (@todo: total) coordinate system (possibly
+                     * after transform to centered origin)
+                     */
+                    HDINLINE PulseFromSpectrum(
+                        typename FieldE::DataBoxType const& dataBoxE,
+                        DataSpace<simDim> const& superCellToLocalOriginCellOffset,
+                        DataSpace<simDim> const& offsetToTotalDomain,
+                        float3_X const& elong,
+                        float_X const phase,
+                        uint32_t const currentStep)
+                        : BaseFunctor(dataBoxE, superCellToLocalOriginCellOffset, offsetToTotalDomain, elong)
+                        , m_phase(phase)
+                        , m_currentStep(currentStep)
+                    {
+                    }
+
+                    /** device side manipulation for init plane (transversal)
```

Comment:
Is this accurate? This operator appears to do the whole thing

Reply:
Yes, it's just outdated in all profiles. The whole thing I meant the whole laser calculation and application